node/src/components/storage.rs

```diff
@@ -340,6 +340,38 @@ impl Storage {
         }
     }
 
+    /// Reads from the state storage DB.
+    /// If key is non-empty, returns bytes from under the key. Otherwise returns `Ok(None)`.
+    /// May also fail with storage errors.
+    #[cfg(not(feature = "fast-sync"))]
+    pub(crate) fn read_state_store<K>(&self, key: &K) -> Result<Option<Vec<u8>>, Error>
+    where
+        K: AsRef<[u8]>,
+    {
+        let txn = self.env.begin_ro_txn()?;
+        let bytes = match txn.get(self.state_store_db, &key) {
+            Ok(&[]) => None,
```

Comment:
It seems like `deserialize_state()` in linear_chain_sync should be handling this case.  How would we even get an empty slice stored there?

Reply:
Maybe I can remove it now. My idea was that if we store an empty vector then we would be effectively removing it from the DB. I never checked that there's a `del` function available. I should use it instead.

Reply:
Removed in [`e0688b5` (#904)](https://github.com/CasperLabs/casper-node/pull/904/commits/e0688b51ed7c2aaff51770152796b4578e1e3268).